packages/SwingSet/src/kernel/vatManager/supervisor-subprocess-xsnap.js

```diff
@@ -243,6 +246,42 @@ function makeWorker(port) {
       meterControl,
     });
 
+    /** @param {string} dst */
+    const makeLogMaker = dst => {
+      /** @param {string} level */
+      const makeLog = level => {
+        // Capture the `console.log`, etc.'s `printAll` function.
+        const printAll = console[level];
+        assert.typeof(printAll, 'function');
+        const portSendingPrinter = (...args) => {
+          port.send([dst, level, ...args]);
+        };
+        return (...args) => {
+          // We have to dynamically wrap the consensus mode so that it can change
+          // during the lifetime of the supervisor (which when snapshotting, is
+          // restored to its current heap across restarts, not actually stopping
+          // until the vat is terminated).
+          if (currentConsensusMode) {
+            return;
+          }
+
+          // Use the causal console, but output to the port.
+          //
+          // FIXME: This is a hack until the start compartment can create
+          // Console objects that log to a different destination than
+          // `globalThis.console`.
+          const { print: savePrinter } = globalThis;
+          try {
+            globalThis.print = portSendingPrinter;
+            printAll(...args);
+          } finally {
+            globalThis.print = savePrinter;
```

Comment:
ok, when `printAll()` causes some random object's `toString` to be invoked and that does more console.log-ing, this structure should unwind the reentrancy properly, nice